draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1137,18 +1143,76 @@ SimulationMarker = {
 ~~~
 {: #simulation-marker-def title="SimulationMarker definition"}
 
-# Event definition guidelines
+# Event extensibility {#event-extensibility}
+
+The main qlog schema defined by this document describes log and trace formats
+that have related events. A minimal set of common events are presented in
+{{events}}. It is expected that logging is extended with specific, per-protocol
+event definitions that specify the name (category + type) and data needed for
+each individual event. Examples include the QUIC event definitions {{QLOG-QUIC}}
+and HTTP/3 event definitions {{QLOG-H3}}.
+
+New event definitions SHOULD follow the guidance in this section, which provides
+consistency that makes it easier for qlog implementers to support a wide variety
+of protocols.
+
+## Extended Event Schema
+
+New event definitions SHOULD be part of an extension schema, using the
+annotations and concepts presented in this section.
+
+An extension schema defines one or more qlog event categories that can contain
+multiple event definitions. Each category MUST have a globally unique category
+identifier. Each event MUST only belong to a single category. Per
+{{{#name-field}}}, event names are the concatenation of category and type.
+Therefore, these requirements ensure there are no ambiguities when multiple
+extension schema are in use.
```

Comment:
I understand what you're saying because I have context, and it's an important thing to highlight, BUT it's at the wrong place in the text... new readers don't know why this is important because you haven't talked about the URIs yet (and how event names tie into those).

Reply:
This section is all about how to extend qlog with more concrete event types. The URI stuff is ancilliary to that.

We need to start with clear requirements on what extensions can and can't do. 

Events have a name that conveys the category and type. When event names are not unique, ambiguitiy can cause tooling to fail. The language here is normative to avoid that class of problem. This problem exists even if we weren't trying to invent a way to indicate what an endpoint was generating e.g. - if we both invent a foobar:frame_sent event type with wildly different data formats.

Reply:
I agree that we have to have clear requirements. I'm mainly saying that this specific text either needs to move (original suggestion) or needs to be explained better in-place (which seems your suggestion) to really make sense for the reader.

I'm guessing this will be easily solved if we clearly explain this up front:

schema namespace > category identifier > event type 

and then it's easier to use that terminology to explain this restriction.